jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -519,3 +526,36 @@ def _live_update(self, event={}):
         for mark in self.marks.values():
             mark.update_xy(sp.spectral_axis.value, sp.flux.value)
             mark.visible = True
+
+    def translate_units(self, spectral_cube, collapsed_spec):
+        if collapsed_spec._unit == u.MJy / u.sr:
+            # MJy to Jy
+            collapsed_spec._data *= 10e6
+
+            # spectral_cube.meta['_pixel_scale_factor'] / 1, where 1 = default scale factor
+            collapsed_spec._data *= spectral_cube.meta['_pixel_scale_factor']
+
+            # per https://jwst-docs.stsci.edu/jwst-near-infrared-camera/nircam-performance/nircam-absolute-flux-calibration-and-zeropoints # noqa
+            # MJy/sr is default, to measure photometry in MJy, convert the image to units
+            # of MJy/pixel by multiplying by the average area of a pixel in sr, a constant
+            # provided in the FITS header keyword PIXAR_SR
+            collapsed_spec._data *= spectral_cube.meta['PIXAR_SR']
```

Comment:
should this, and the original unit, be absorbed into the scale factor so we can _just_ multiply or divide by a quantity?